feeluown/containers/player_control_panel.py

```diff
@@ -27,9 +27,65 @@ class SongBriefLabel(QLabel):
 
     def __init__(self, app):
         super().__init__(text=self.default_text, parent=None)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Preferred)
+
         self._app = app
+        self._timer = QTimer()
+        self._txt = self._raw_text = self.default_text
+        self._font_metrics = QFontMetrics(QApplication.font())
+        self._text_rect = self._font_metrics.boundingRect(self._raw_text)
+        # text's position, keep changing to make text roll
+        self._pos = 0
+
+        self._timer.timeout.connect(self.change_text_position)
+
         self._fetching_artists = False
 
+    def change_text_position(self):
+        if not self.parent().isVisible():
+            self._timer.stop()
+            self._pos = 0
+            return
+        if self._text_rect.width() + self._pos > 0:
+            # control the speed of rolling
+            self._pos -= 5
+        else:
+            self._pos = self.width()
+        self.update()
+
+    def setText(self, text):
+        self._txt = self._raw_text = text
+        self._text_rect = self._font_metrics.boundingRect(self._raw_text)
+        self._pos = 0
+        self.update()
+
+    def enterEvent(self, event):
+        if self._txt != self._raw_text:
+            # decrease to make rolling more fluent
+            self._timer.start(150)
+
+    def leaveEvent(self, event):
+        self._timer.stop()
+        self._pos = 0
+        self.update()
+
+    def paintEvent(self, event):
+        painter = QPainter(self)
+        painter.setFont(QApplication.font())
+        painter.setPen(self._app.palette().color(self._app.palette().Text))
+
+        if self._timer.isActive():
+            self._txt = self._raw_text
+        else:
+            self._txt = self._font_metrics.elidedText(
+                self._raw_text, Qt.ElideRight, self.width())
+
+        painter.drawText(
+            QRect(self._pos, 0, self.width()-self._pos, self.height()),
```

Comment:
LGTM

adding space between operators is a preferred coding style.
please fix this when merging.